docs/ga-roadmap.md

```diff
@@ -0,0 +1,173 @@
+them will be minimal: `pdata` is auto-generated so all changes that we do for trace will
+apply to all of them, `consumer` is minimal interface, `component` as well.
+
+Metrics components such as (`receiver/prometheus`, `exporter/prometheusremotewrite`) are
+explicitly left out because metrics data model is not complete, but if that work finishes,
+we can discuss and add them to the Phase 3, or later.
+
+### Phase 1
+
+**Tentative Date:** 2021-03-31
+
+**KR:** At the end of this phase the Collector’s core API/SPI will be marked as Stable.
+
+At the end of this phase we want to achieve core APIs stability. This will allow developers
+to implement custom components and extend the collector will be marked as stable.
+The complete list of the packages/modules will be finalized during the first action item of
+this phase, but the tentative list is:
+
+* `consumer`
+  * Official internal data model `pdata`.
+  * Interfaces and utils to build a Consumer for (trace, metrics, logs).
+* `config`
+  * Core `config` including service definition, component definition will be stabilized.
+  * To be determined which config helpers will be marked as stable (e.g. configgrpc, etc.).
+* `component`
+  * Interfaces and utils to build a Collector component (receiver, processor, exporter, extension).
+* `obsreport`
+  * Focus on the public API of this package. It is out of scope to ensure stability for the
+  metrics emitted (focus in phase 2).
+* `service`
+  * Public API to construct a OpenTelemetry Collector Service.
+
+**Action Items:**
+
+* Create a new milestone for this phase, create issues for all the other action items and add
+them to the milestone.
+* Agreement on all packages/modules that will be marked as stable during this phase.
+* Write a version doc as per [version and stability document](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md).
+  * Previously it was discussed that for the Collector it is fine to release stable golang modules
+  that contain APIs marked as experimental.
+  * Define status schema (experimental/stable), what are they applicable to every module.
+* Investigate if splitting into smaller, more granular, modules is possible.
+  * Define the modules schema, try to not break everyone.
+  See [here](https://github.com/golang/go/wiki/Modules#is-it-possible-to-add-a-module-to-a-multi-module-repository).
+  * Investigate how can we release multiple golang modules from the same repo, without asking
+  people that consume them to use a replace statement. See problem in contrib.
+  * Investigate how to release testonly utils.
+* Review all public APIs and godocs for modules that we want to release in this phase.
+  * Fix all critical issues, and remove unnecessary (“When in doubt leave it out”) public APIs.
+  * Remove all already deprecated code from the stable modules.
+* Transition to opentelemetry-go trace library from opencensus?
```

Comment:
Could this wait given the opentelemetry-go is not finalized either?

Reply:
It is a question mark for myself as you can see in the comment. The only part that I don't know the answer is if a custom component needs to do custom tracing should we recommend OpenCensus or Otel? Because if we recommend OpenCensus that may be a breaking change when switching to Otel.

Also this will help Otel-go identifying potential issues, so it may worth doing.

Conclusion: Not sure, but if we don't use it I would at least put a statement that custom tracing support for custom components is not supported in this version.

Reply:
We may need to investigate if we can wrap OC or Otel traces and metrics in a way that hides them as implementation detail and expose the wrapper as an API for components to use.

Reply:
> We may need to investigate if we can wrap OC or Otel traces and metrics in a way that hides them as implementation detail and expose the wrapper as an API for components to use.

Is that how the OpenTelemetry project as a whole recommends? We are a highly visible, and potentially one of the first, users of the OpenTelemetry instrumentation libraries and we should set an example. If the project as a whole recommends using the APIs directly, that's what we should be doing and advocating for.

Reply:
I know but there is a race between when otel go will be ready and we plan to be ready

Reply:
There is no stable metrics Otel API yet, so it can't be recommended. For traces, I think you are right, assuming opentelemetry-go trace will be stable soon before Collector GAs.

Wrapping may still be beneficial, not for the purpose of hiding breaking changes but to provide convenience / helper methods to components. I think it is worth researching.